peering/peering.go

```diff
@@ -126,6 +126,9 @@ func (ph *peerHandler) reconnect() {
 		if ph.reconnectTimer != nil {
 			// Only counts if the reconnectTimer still exists. If not, a
 			// connection _was_ somehow established.
+			if !ph.reconnectTimer.Stop() {
+				<-ph.reconnectTimer.C
```

Comment:
This will deadlock and leak a goroutine each time. This timer is created with `AfterFunc` which doesn't even create a channel (or send on it), so the read on the nil channel will hang forever.

In general, be _very_ careful when working with timers.